src/units/abilities.cpp

```diff
@@ -393,20 +381,20 @@ namespace {
 	}
 }//anonymous namespace
 
-unit::recursion_guard unit::update_variables_recursion() const
+unit::recursion_guard unit::update_variables_recursion(const config& ability) const
 {
-	if(num_recursion_ < UNIT_RECURSION_LIMIT) {
-		return recursion_guard(*this);
+	if(utils::contains(open_queries_, ability)) {
```

Comment:
Since this now returns a default instance, is it possible to make the function returns `utils::optional<recursion_guard>`? That is, does a default instance mean anything or will it always evaluate to false in `operator bool`>

Reply:
i don't understand your question, if unit::recursion_guard filter_lock exist(while cfg checked not already present in open_queries_), check continue normally, and for unit::recursion_guard systemi don't know yet for reply to your question.

Reply:
I remember looking at `std::optional<recursion_guard>` and finding it wasn't useful. IIRC, using `std::optional` required `recursion_guard` to support `std::move`. Once the class supports a moved-from state (implemented by `parent == nullptr`), it's logical to use that moved-from state instead of wrapping the class in a `std::optional`.